cms/djangoapps/models/settings/course_details.py

```diff
@@ -188,6 +192,12 @@ def update_from_json(cls, course_key, jsondict, user):
             descriptor.language = jsondict['language']
             dirty = True
 
+        if (SelfPacedConfiguration.current().enabled
+                and 'self_paced' in jsondict
+                and jsondict['self_paced'] != descriptor.self_paced):
+            descriptor.self_paced = jsondict['self_paced']
+            dirty = True
```

Comment:
I think this should be conditional, just to be on the safe side -- if self-paced courses are disabled, it doesn't make sense to be able to be able to set a course to self-paced through this endpoint (even if the UI component isn't shown). Line 197 does the check to see if the value actually changes, so `dirty = True` should be set.